cfep-9-migrator-file.md

```diff
@@ -0,0 +1,209 @@
+# @mariusvniekerk's proposal
+
+This is a general sketch of what I'd like migrators to be able to achieve.  
+Effectively this means we can decompose all of our migrations into a small set of primitives 
+that we can apply to generate a resulting desired target.
+
+----
+
+Given a feedstock with the following pinnings
+
+```yaml
+# Existing pins
+a:
+  - 1.5
+b:
+  - 1.2
+someflag:
+  - disabled
+```
+
+There are four classes of pinning we can do in general.
+
+## 1. Simple Additive Version Migrations
+
+<table>
+<tr>
+<td>
+
+```yaml
+# Existing pins
+a:
+  - 1.5
+b:
+  - 1.2
+someflag:
+  - disabled
+```
+</td>
+<td> with </td>
+<td>
+
+```yaml
+# migrator
+__migrator: kind: version
+a:
+  - 1.6
+b:
+  - 1.1
+```
+</td>
+<td> -> </td>
+<td>
+
+```yaml
+# resulting pins
+a:
+  - 1.6
+b:
+  - 1.2
+someflag:
+  - disabled
+```
+</td>
+</tr>
+</table>
+
+These are the most common version migrations and encompass things like moving the pinnings for openssl etc.
+
+Note that the version of `b` was NOT decreased.  This is intentional as we don't wish to downgrade accidentally.  In practice many migrators could be in flight and may affect the same version.  In this case the higher version should always 
+win.  If we want to downgrade there is a mechanism but it is purposefully clunky and bad.
```

Comment:
Hmmmm. This answers my question above at some level. However I’d rather not personally be in the weeds on package versions. Changes to the global pinnings are fundamentally git file diffs.